public/coffee/main/event.coffee

```diff
@@ -1,19 +1,50 @@
 define [
 	"base"
+	"modules/localStorage"
 ], (App) ->
+	CACHE_KEY = "countlyEvents"
+
+	App.factory "event_tracking", (localStorage) ->
+		_getEventCache = () -> 
+			eventCache = localStorage CACHE_KEY
+
+			# Initialize as an empy object if the event cache is still empty.
+			if !eventCache?
+				eventCache = {}
+				localStorage CACHE_KEY, eventCache 
+
+			return eventCache
+
+		_eventInCache = (key) ->
+			curCache = _getEventCache()
+
+			if (curCache.hasOwnProperty key) 
```

Comment:
out of interest what does hasOwnProperty add over doing

curCache[key] || false